worker/codalabworker/local_run/local_run_state.py

```diff
@@ -0,0 +1,277 @@
+            status_message = 'Pulling docker image: ' + (image_state.message or docker_image)
+            return run_state._replace(run_status=status_message)
+        elif image_state.stage == DependencyStage.FAILED:
+            # Failed to pull image; -> CLEANING
+            run_state.info['failure_message'] = image_state.message
+            return run_state._replace(stage=LocalRunStage.CLEANING, info=run_state.info)
+
+        # All dependencies ready! Set up directories, symlinks and container. Start container.
+        # 1) Set up a directory to store the bundle.
+        remove_path(run_state.bundle_path)
+        os.mkdir(run_state.bundle_path)
+
+        # 2) Set up symlinks
+        bundle_uuid = run_state.bundle['uuid']
+        dependencies = []
+        docker_dependencies_path = '/' + bundle_uuid + '_dependencies'
+        for dep in run_state.bundle['dependencies']:
+            child_path = os.path.normpath(os.path.join(run_state.bundle_path, dep['child_path']))
+            if not child_path.startswith(run_state.bundle_path):
+                raise Exception('Invalid key for dependency: %s' % (dep['child_path']))
+
+            dependency_path = self._run_manager.dependency_manager.get(
+                        bundle_uuid,
+                        (dep['parent_uuid'], dep['parent_path'])).path
+            dependency_path = os.path.join(self._run_manager.dependencies_dir, dependency_path)
+
+            docker_dependency_path = os.path.join(docker_dependencies_path, dep['child_path'])
+
+            os.symlink(docker_dependency_path, child_path)
+            # These are turned into docker volume bindings like:
+            #   dependency_path:docker_dependency_path:ro
+            dependencies.append((dependency_path, docker_dependency_path))
+
+        # 3) Set up container
+        if run_state.resources['request_network']:
+            docker_network = self._run_manager.docker_network_external_name
+        else:
+            docker_network = self._run_manager.docker_network_internal_name
+
+        try:
+            cpuset, gpuset = self._run_manager.assign_cpu_and_gpu_sets(
+                run_state.resources['request_cpus'], run_state.resources['request_gpus'])
+        except:
+            run_state.info['failure_message'] = "Cannot assign enough resources"
+            return run_state._replace(stage=LocalRunStage.CLEANING, info=run_state.info)
+
+        # 4) Start container
+        container_id = self._run_manager.docker.start_container(
+            run_state.bundle_path, bundle_uuid, run_state.bundle['command'],
+            run_state.resources['docker_image'], docker_network, dependencies,
+            cpuset, gpuset, run_state.resources['request_memory']
+        )
+
+        digest = self._run_manager.docker.get_image_repo_digest(run_state.resources['docker_image'])
+
+        return run_state._replace(stage=LocalRunStage.RUNNING, run_status='Running',
```

Comment:
make run_status a bit longer so it looks more like a status message than a stage.